### PR TITLE
Avoid running hasMetaBoxes on each subscribe

### PR DESCRIPTION
**Pull request by youknowriad**
While monitoring the typing performance, I noticed that in the duraction of the key press event (53ms), 2 or 3ms is spent on calling `hasMetaboxes` selector. In this PR, I'm managing to tweak the side effect a little bit to avoid calling this selector on each subscribe.

While if we're trying to be pure, we should call the selector on each subscribe, in this particular use-cases where this is a legacy API that is only initialized at rendering, I think it's fine to optimize this way.

### CODE REVIEW

**Comment by aduth** [comment]
> 2 or 3ms is spent on calling `hasMetaboxes` selector.

Is there a sense of why it has a noted impact? Curious if there's something problematic of the underlying selector.

I'd probably agree if we can be certain it's initialized once, it would be fine to move it out of the `subscribe`. Would be good to have a reference to this initialization.

File: packages/edit-post/src/store/effects.js [hunk 1]
```diff
@@ -45,11 +45,11 @@ const effects = {
 
 		let wasSavingPost = select( 'core/editor' ).isSavingPost();
 		let wasAutosavingPost = select( 'core/editor' ).isAutosavingPost();
+		const hasActiveMetaBoxes = select( 'core/edit-post' ).hasMetaBoxes();
```
Comment (aduth) [inline packages/edit-post/src/store/effects.js]:
```suggestion
		
		// Meta boxes are initialized once at page load. It is not necessary to
		// account for updates on each state change.
		//
		// See: https://github.com/WordPress/WordPress/blob/5.1.1/wp-admin/includes/post.php#L2307-L2309
		const hasActiveMetaBoxes = select( 'core/edit-post' ).hasMetaBoxes();
```

**Review by aduth — APPROVED** [review]

**Comment by ellatrix** [comment]
Looks like this was merged with failing tests.

**Comment by youknowriad** [comment]
yep sorry about that missed it, I included the suggestion above and thought was harmless cause it was just a comment :)

**Pull request merged**